xla/service/gpu/nccl_clique_key.h

```diff
@@ -27,10 +27,11 @@ limitations under the License.
 #include "absl/status/statusor.h"
 #include "absl/types/span.h"
 #include "xla/service/global_device_id.h"
+#include "tsl/lib/gtl/int_type.h"
 
 namespace xla::gpu {
 
-TSL_LIB_GTL_DEFINE_INT_TYPE(NcclStreamId, uint64_t);
+TSL_LIB_GTL_DEFINE_INT_TYPE(ExecutionStreamId, uint64_t);
```

Comment:
`ExecutionStreamId` and `NcclStreamId` should stay separate. Execution streams correspond to "physical" CUDA streams that we launch work, and NcclStream are "virtual" to distinguish different cliques, and same nccl stream can be mapped to N different excecution streams at different point in time.

By making nccl clique depending on excecution stream we can potentially creates tens (hundreds) of redundant cliques.